pkg/api/pod/util.go

```diff
@@ -383,6 +383,14 @@ func dropDisabledFields(
 		})
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StartupProbe) && !startupProbeInUse(oldPodSpec) {
+		// drop startupProbe from all containers if the feature is disabled
+		VisitContainers(podSpec, func(c *api.Container) bool {
```

Comment:
Should ephemeral container be excluded ?

Reply:
that's a good point, will open a follow-up issue to track usage of VisitContainers in *InUse methods

Reply:
opened https://github.com/kubernetes/kubernetes/issues/82158